### PR TITLE
GSR Booking Credentials

### PR DESCRIPTION
**Pull request by younglee327**
- New model and serializer for GSR Booking Credentials
- GET method for getting a user's GSR Booking Credentials (`gsr_booking_credentials`)
- POST methods for saving Session ID and school email (`save_session_id` and `save_email`)
- Updated documentation

### CODE REVIEW

File: gsr_booking/models.py [hunk 1]
```diff
@@ -82,3 +85,28 @@ def save(self, *args, **kwargs):
...
+
+    # Expiration date of the Session ID
+    expiration_date = models.DateTimeField("Session ID expiration date", null=True)
+
+    # When Session ID or email was last updated
+    # TODO: May want to differentiate email and Session ID added times
+    date_updated = models.DateTimeField(auto_now_add=True)
+
+    # For LibCal, school emails are used instead
+    email = models.CharField("school email", max_length=255, unique=True, null=True)
```
[the remark this answers is not in the document]
Comment (r3khaan) [reply gsr_booking/models.py]: Just curious, how do we get the `user.email` field?

File: gsr_booking/views.py [hunk 2]
```diff
@@ -64,6 +65,89 @@ def search(self, request):
...
+        # Else create a new credentials object and associate it with the user
+        else:
+            GSRBookingCredentials.objects.create(
+                user=user, session_id=session_id, expiration_date=expiration_date
+            )
+
+        return Response({"success": True})
+    
+    @action(detail=True, methods=["post"])
+    def save_email(self, request, username=None):
```
[the remark this answers is not in the document]
Comment (r3khaan) [reply gsr_booking/views.py]: Just curious, is there a `RetrieveUpdateAPIView` for `ModelViewSet`? I see that everything we are using so far is some kind of `ModelViewSet`, so I'm tempted to use that as opposed to a `GenericView`.

And also, if I did use a `ModelViewSet`, I'm guessing I would do the validation in a Serializer like you did for group-booking right? And nothing else to change right?

File: gsr_booking/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (davish) [reply gsr_booking/models.py]: The school made a Directory API for us to access emails given we have a students penn ID number, which we get through the login process

File: gsr_booking/views.py [hunk 3]
```diff
@@ -113,6 +114,31 @@ def search(self, request):
...
+class GSRBookingCredentialsViewSet(generics.RetrieveUpdateAPIView, generics.DestroyAPIView):
+    permission_classes = [IsAuthenticated]
+    authentication_classes = (CsrfExemptSessionAuthentication, BasicAuthentication)
+    serializer_class = GSRBookingCredentialsSerializer
+
+    def get_object(self):
+        if not self.request.user.is_authenticated:
+            return GSRBookingCredentials.objects.none()
+        try:
+            return GSRBookingCredentials.objects.get(pk=self.request.user.pk)
```
[the remark this answers is not in the document]
Comment (r3khaan) [reply gsr_booking/views.py]: Will do! I had actually set the `user` field to the primary key of the `GSRBookingCredentials` object, but guessing thats not the Django way?

**Pull request merged**